src/source.rs

```diff
@@ -39,3 +35,48 @@ impl JwksSource for WebSource {
         Ok(keys)
     }
 }
+
+pub struct WebSourceBuilder {
+    client_builder: reqwest::ClientBuilder,
+    url: Url,
+    timeout_opt: Option<Duration>,
+    connect_timeout_opt: Option<Duration>,
+}
+
+impl WebSourceBuilder {
+    pub fn new(url: Url) -> Self {
```

Comment:
I prefer having a `WebSource::builder()` function, instead of having to know the presence of the builder struct. WDYT?

Reply:
Good point!